src/nft/components/details/detailsV2/TraitRow.tsx

```diff
@@ -27,21 +45,32 @@ const TraitRowValue = styled(ThemedText.BodySmall)<{ $flex?: number; $justifyCon
   justify-content: ${({ $justifyContent }) => $justifyContent};
 `
 
-export const TraitRow = ({ trait }: { trait: Trait }) => {
+export const TraitRow = ({ trait, collectionAddress }: { trait: Trait; collectionAddress: string }) => {
+  const [rowHovered, toggleRowHovered] = useReducer((s) => !s, false)
   // TODO: Replace with actual rarity, count, and floor price when BE supports
   // rarity eventually should be number of items with this trait / total number of items, smaller rarity means more rare
-  const randomRarity = Math.random()
+  const randomRarity = useMemo(() => Math.random(), [])
+  const rarityLevel = getRarityLevel(randomRarity)
+  const params = qs.stringify(
+    { traits: [`("${trait.trait_type}","${trait.trait_value}")`] },
+    {
+      arrayFormat: 'comma',
+    }
+  )
   return (
-    <Row padding="12px 18px 12px 0px">
-      <TraitValue>
-        <SubheaderTiny>{trait.trait_type}</SubheaderTiny>
-        <ThemedText.BodyPrimary lineHeight="20px">{trait.trait_value}</ThemedText.BodyPrimary>
-      </TraitValue>
-      <TraitRowValue $flex={2}>{formatEth(randomRarity * 1000)} ETH</TraitRowValue>
-      <TraitRowValue>{Math.round(randomRarity * 10000)}</TraitRowValue>
-      <TraitRowValue $flex={1.5} $justifyContent="flex-end">
-        <RarityGraph trait={trait} rarity={randomRarity} />
-      </TraitRowValue>
-    </Row>
+    <TraitRowLink to={`/nfts/collection/${collectionAddress}?${params}`}>
+      <TraitRowContainer onMouseEnter={toggleRowHovered} onMouseLeave={toggleRowHovered}>
+        <TraitColumnValue>
+          <SubheaderTiny>{trait.trait_type}</SubheaderTiny>
+          <ThemedText.BodyPrimary lineHeight="20px">{trait.trait_value}</ThemedText.BodyPrimary>
+        </TraitColumnValue>
+        <TraitRowValue $flex={2}>{formatEth(randomRarity * 1000)} ETH</TraitRowValue>
+        <TraitRowValue>{Math.round(randomRarity * 10000)}</TraitRowValue>
+        <TraitColumnValue $flex={1.5} $alignItems="flex-end">
+          <SubheaderTiny $color={rarityLevel.color}>&nbsp;{rowHovered && rarityLevel.hoverText}</SubheaderTiny>
```

Comment:
The `&nbsp;` is used to maintain the spacing in the column when the hoverText is not shown